//客户端拿验证token来登陆代理
message login_proxy
{
	required bytes im_account = 1;
	required bytes auth_code = 2;
	required int32 im_uid = 3;
	optional bytes proxy_topic = 4;
}

//错误码
message errorinfo
{
	required  int32 error_code =1;
	optional  bytes error_info =2;
}

//群类型
enum group_type 
{
	unknown_group    = 0; //未知
	business_group   = 1; //业务群
	common_group     = 2; //普通群
	discussion_group = 3; //讨论组
}

//消息体  and longitude
message msg_content
{
	required bytes msg = 1;       //如果为文本，只填msg，如果图片，只填msg(缩略图)，
								  //msg_url(原图),如果为语音，msg为语音路径，如果是名片人昵称

	optional bytes msg_url =2;    //如果发送的是名片则名片人的头像
	optional int32 voice_len =3;  //如果语音，则为语音长度，如果是名片则是名人的id    
	optional double longitude=4;	  //经度
	optional double latitude= 5;      //维度
	optional int32  thumb_width = 6 [ default  = 0]; //缩略图宽度
	optional int32  thumb_hight = 7 [ default  = 0]; //缩略图宽度
	optional bytes  nickname    = 8;
	optional bytes  avatar_url  = 9;
    optional int32  red_pket_id = 10;  // 红包id

}

// 登陆代理回应
message login_proxy_ack
{
	optional int32 sessionid = 1;
	optional errorinfo error = 2;
}

//对于发送者msg_serial，strange_stauts不需要填，接受着的话一定要填
message msgbody
{
	required  msg_content msg = 1;
	optional  bytes msg_serial       = 2; //旧版消息序列号，升级后不再使用
	optional  int64 time             = 3;
	optional  int32 strange_stauts   = 4; //陌生人，第一次会话，1是陌生人第一次
    optional  int64 msg_serial_t     = 5; //消息序列号，时间戳版本
	optional  group_type gtype= 6; //群类型，群聊的时候需要用到
}

//消息回执
message msgbody_ack
{
	required  errorinfo error = 1;
	optional  bytes msg_serial       = 2; //旧版消息序列号，升级后不再使用
	optional  int64 time             = 3;
    optional  int64 msg_serial_t     = 4; //消息序列号，时间戳版本
}
